### ISSUE TITLE
Re-Add Sort/Drag&Drop Blogs

### ISSUE THREAD

**Issue by MinecraftMoments** [AprilSylph/XKit-Rewritten#1153]
### Concept

I have a blog that I newly created, and want to put it up higher on my Tumblr's blog list. Unfortunately, Tumblr Itself has removed this feature from being available. I want to be able to sort my blogs again, and IDK how long Tumblr will take to readd it, if at all.

**Comment by marcustyphoon** [comment]
Tumblr has not removed this! Hover over one of your sideblogs in the account menu and four dots should appear that allow you to drag to reorder. You can't make your main blog not the first one, though.

Are you not seeing this appear?

**Comment by ketherphorbia** [comment]
I am also experiencing this problem. I haven't been able to drag my blog order for at least a few weeks now. I sent Tumblr a support ticket in case there was something wrong with my account, but I haven't received a reply. I've tested it in both Firefox and Edge. None of my Discord friends have experienced this issue.

No idea if the site's layout update impacts anything, either.

**Comment by marcustyphoon** [comment]
Same question to you: Are you not seeing the four dots appear, or are you not able to use them? Do you have more than two blogs?

**Comment by ketherphorbia** [comment]
> Same question to you: Are you not seeing the four dots appear, or are you not able to use them? Do you have more than two blogs?

I'm not seeing the four dots at all. I have more than twenty side blogs, so being able to organize them is... pretty important for me.

**Comment by ketherphorbia** [comment]
> Same question to you: Are you not seeing the four dots appear, or are you not able to use them? Do you have more than two blogs?
![Screenshot 2023-07-12 234902](https://github.com/AprilSylph/XKit-Rewritten/assets/41311509/07fd8d2a-92ba-4774-b0c0-15567c301551)

Screenshot of two side blogs in the settings dropdown, from before the site layout overhaul.

![Screenshot 2023-07-22 181539](https://github.com/AprilSylph/XKit-Rewritten/assets/41311509/ab862b3a-2810-4f56-b9fd-359bb54e50ad)

Screenshot of two side blogs in the settings sidebar, as of right now after the site layout changed.

**Comment by marcustyphoon** [comment]
The buttons in question don't appear until you mouse over a blog (and won't appear on your main blog, as it can't be moved down). Does this occur for you?

<img width="230" src="https://github.com/AprilSylph/XKit-Rewritten/assets/8336245/5e8b4c7b-a3d4-4b2f-8d1d-3e2fbb52ab51">

**Comment by ketherphorbia** [comment]
> The buttons in question don't appear until you mouse over a blog (and won't appear on your main blog, as it can't be moved down). Does this occur for you?
> <img alt="" width="230" src="https://user-images.githubusercontent.com/8336245/255356762-5e8b4c7b-a3d4-4b2f-8d1d-3e2fbb52ab51.png">

I am hovering over it on the second image there. I have tried hovering. It does not appear either way.

**Comment by ketherphorbia** [comment]
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/41311509/9dac5428-9ada-4a49-89ac-5963d3c3b841)
Okay, this is a strange discovery. I still have the grabby-dots button in the queue screen ??? So my account still at least has that functionality. It's just bafflingly not present in any listing of my blogs.

**Comment by AprilSylph** [comment]
The four dots won't appear if your browser has decided that your primary pointing device is a touchscreen. If you have a hybrid device, try restarting your browser with the mouse attached/enabled?

**Comment by marcustyphoon** [comment]
That seems like an obvious Tumblr bug; shouldn't the four dots **always** appear if you have a touchscreen?

**Comment by AprilSylph** [comment]
Maybe... it's hard to reconcile the need to make both the drag handle and expand/collapse button big enough to be tap targets with not hiding even more of the blog info. But I guess form shouldn't impede function...

**Comment by marcustyphoon** [comment]
Yeah, I mean—I would say something about the blog dropdown itself being redundant but I guess that's a separate conversation :P

**Comment by AprilSylph** [comment]
The blog dropdown is not redundant. There is no right rail on mobile.

**Comment by ketherphorbia** [comment]
> The four dots won't appear if your browser has decided that your primary pointing device is a touchscreen. If you have a hybrid device, try restarting your browser with the mouse attached/enabled?

Okay, so this is wild. To start, you found a solution to my issue.

I do not have a touchscreen laptop. It has a trackpad, but the screen is "normal." I've had it several years now, and I've always been able to drag to change my blog order, up until about a month or so ago.

I closed Firefox, plugged in an external mouse, and restarted the browser. The grabby-dots button is there now and I can drag to reorder my blogs?

Thank you so much. Hopefully I won't have to plug in my external mouse every time I need to do this, but the important thing is that I *can* do this.

**Comment by ketherphorbia** [comment]
One other thing, too. I tested it for science. Had to know.

I plugged in my Wacom. The grabby-dots appear both with touch turned on using my finger, and also with touch turned off using a stylus.

So the issue is definitely my trackpad. I'm about to check my Device Manager to make sure the drivers didn't get screwy somehow. That seems like the most likely explanation, given what does and doesn't produce abnormal behavior.

Thank you again!!
